src/main/java/com/johnstarich/moviematcher/models/User.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    public User(ObjectId _id, String email, String first_name, String last_name, String password) {
+        this._id = _id;
+        this.email = email;
+        this.first_name = first_name;
+        this.last_name = last_name;
+        this.friends = new ArrayList<>(0);
+        this.groups = new ArrayList<>(0);
+        this.password = password;
+    }
+
+    public User(ObjectId _id, String email, String first_name, String last_name) {
+        this._id = _id;
+        this.email = email;
+        this.first_name = first_name;
+        this.last_name = last_name;
+        this.friends = new ArrayList<>(0);
+        this.groups = new ArrayList<>(0);
+        this.password = null;
+    }
+
+    @Override
+    public boolean equals(Object o) { return o == this || o instanceof User && ((User) o)._id == _id && ((User) o).email == email; }
+
+    @Override
+    public int hashCode() {
+        return _id.hashCode();
+    }
+
+    public User load() {
+        Document user = getCollection().find(eq("_id", _id)).first();
+        Gson gson = new GsonBuilder().create();
+        return gson.fromJson(user.toJson(), User.class);
+    }
+
+    public User register(String password) throws HttpException {
+        if(this.load().equals(this)) { //need to test this, too see if it actually works!! ...
+            throw new HttpException(HttpStatus.BAD_REQUEST);
+        }
+
+        User u = new User(_id, email, first_name, last_name, friends, groups, BCrypt.hashpw(password, BCrypt.gensalt()));
+        u.insert();
+        return u;
+    }
+
+    /* not sure how to add friends or groups to mongoDB since they are lists*/
+    /* will create the collection, once we insert our first user */
+    public void insert() {
+        getCollection().insertOne(
+                new Document("_id", _id).append("email" , email).append("first_name" , first_name)
+                        .append("last_name", last_name).append("password", password)
+                        .append("friends", friends).append("groups", groups)
+        );
+    }
+
+    public static List<User> search(String query) {
+        AggregateIterable<Document> iterable = getCollection().aggregate(
+            asList(
+                new Document("$match", new Document("$text", new Document("$search", query))),
+                new Document(
+                    "$project",
+                    new Document("email", true)
+                        .append("first_name", true)
+                        .append("last_name", true)
+                        .append("friends", true)
+                        .append("groups", true)
+                        .append("password", true)
+                        .append("score", new Document("$meta", "textScore"))
+                ),
+                new Document("$sort", new Document("score",-1))
+            )
+        );
+
+        ArrayList<User> queryResults = new ArrayList<User>();
+        Gson gson = new GsonBuilder().create();
```

Comment:
Make this a static instance inside the class so it isn't created on every method call